module/handler/fast_forward.py

```diff
@@ -190,27 +185,86 @@ def triggered_map_stop(self):
+                on or off
+            check_button (Button):
+                button to check before attempting to click
+            box_button (Button):
+                button to click and image color count against
+            skip_first_screenshot (bool):
+                namesake
+
+        Returns:
+            bool:
+                True if detected having set correctly
+                False can occur for 2 reasons either
+                assets insufficient to detect properly
+                or 2x book setting is absent
+
+        """
+        clicked_timeout = Timer(3, count=6)
+        clicked_threshold = 3
+        while 1:
+            if skip_first_screenshot:
+                skip_first_screenshot = False
+            else:
+                self.device.screenshot()
+
+            if clicked_threshold < 0:
+                break
+
+            if clicked_timeout.reached():
+                if self.appear(check_button):
+                    enabled = self.image_color_count(box_button, color=(156, 255, 82), threshold=221, count=20)
+                    if ((status == 'on' and enabled) or (status == 'off' and not enabled)):
+                        return True
+                    if ((status == 'on' and not enabled) or (status == 'off' and enabled)):
+                        self.device.click(box_button)
+
+                clicked_timeout.reset()
+                clicked_threshold -= 1
+
+        logger.warn(f'Wait time has expired; Cannot set 2x book setting')
+        return False
+
     def handle_2x_book_setting(self, mode='prep'):
         """
         Handles 2x book setting if applicable
+
+        Args:
+            mode (string):
+                prep or auto, assume auto if not prep
+
+        Returns:
+            bool:
+                If handled to completion
         """
         if not hasattr(self, 'emotion'):
             logger.info('Emotion instance not loaded, cannot handle 2x book setting')
             return False
 
+        logger.info(f'Handling 2x book setting, mode={mode}.')
         if mode == 'prep':
-            book = book_prep
+            book_check = BOOK_CHECK_PREP
+            book_box = BOOK_BOX_PREP
         else:
-            book = book_auto
+            book_check = BOOK_CHECK_AUTO
+            book_box = BOOK_BOX_AUTO
 
-        if not book.appear(main=self):
-            logger.info(f'No 2x book option, mode={mode}.')
+        status = 'on' if self.map_is_2x_book else 'off'
+        if self._set_2x_book_status(status, book_check, book_box):
+            self.emotion.map_is_2x_book = self.map_is_2x_book
+        else:
             self.map_is_2x_book = False
             self.emotion.map_is_2x_book = self.map_is_2x_book
-            return False
-
-        self.emotion.map_is_2x_book = self.map_is_2x_book
-        status = 'on' if self.map_is_2x_book else 'off'
 
-        book.set(status=status, main=self)
         self.ensure_no_info_bar()
+        return True

```

Comment:
Use `self.handle_info_bar()` instead of `self.ensure_no_info_bar()`.
Info bar only appears when option is changed. If option is correct, handle_info_bar can save some time.